base/compiler/ssair/passes.jl

```diff
@@ -446,9 +446,10 @@ function lift_arg!(
     end
     if isa(lifted, GlobalRef) || isa(lifted, Expr)
         lifted = insert_node!(compact, leaf, effect_free(NewInstruction(lifted, argextype(lifted, compact))))
-        compact[leaf] = nothing
+        leaf_id = isa(leaf, AnySSAValue) ? leaf.id : leaf
```

Comment:
I don't believe this is correct. If it's an old ssa value, it needs to be renamed. This might work 90% of the time if the old and new ssa value numbers happen to agree, but it'll definitely fail otherwise.